astropy/io/fits/file.py

```diff
@@ -358,6 +358,7 @@ def readarray(self, size=None, offset=0, dtype=np.uint8, shape=None):
                         # is to open the file in mode='denywrite', which at
                         # least allows the file to be opened even if the
                         # resulting arrays will be truly read-only.
+
                         if exc.errno == errno.ENOMEM and self.mode == "readonly":
```

Comment:
I think you can simply expand this `if` statement as `if (exc.errno == errno.ENOMEM or exc.winderror == 1455) and ...`

I think the error message is clear enough also for the windows error and the solution is the same.
(But do update the comment above, merging with what you added below.)